src/XenCert/StorageHandler.py

```diff
@@ -1836,20 +1835,74 @@ def deleteVdiFromMetadata(self, vdi_uuid):
     
     def Probe_SR(self):
         return ''
-        
-class StorageHandlerISCSI(StorageHandler):
+
+
+class BlockStorageHandler(StorageHandler):
+
+    def __init__(self, storage_handler):
+        super(BlockStorageHandler, self).__init__(storage_handler)
+        self.gfs2_handler = StorageHandlerGFS2(self)
+        self.enabled_clustering = False
+
+    def Gfs2Supported(self):
+        pools = self.session.xenapi.pool.get_all_records()
+        pool = [pools[k] for k in pools][0]
+        if not 'restrict_corosync' in pool['restrictions']:
+            return False
+
+        if not self.session.xenapi.Cluster.get_all():
+            # cluster not enabled
+            Print("Enabling clustering")
+            management_pifs = [ref for (ref,pif) in
+                               self.session.xenapi.PIF.get_all_records().items()
+                               if pif['management']]
+            [self.session.xenapi.PIF.set_disallow_unplug(pif, True) for
+             pif in management_pifs]
+
+            management_network = self.session.xenapi.PIF.get_record(
+                management_pifs[0])['network']
+            self.cluster = self.session.xenapi.Cluster.pool_create(
+                management_network, 'corosync', 1.0, 0.65)
+            self.enabled_clustering = True
+
+        return True
+
+    def DisableClustering(self):
+        """
+        If we enabled clustering disable it afterwards
+        """
+        if self.enabled_clustering:
+            Print("Test enabled clustering, disabling at end")
+            self.session.xenapi.Cluster.pool_destroy(self.cluster)
+
+    def ControlPathStressTests(self):
+        (retValControl, checkPointsControl, totalCheckPointsControl) = \
+            super(BlockStorageHandler, self).ControlPathStressTests()
+
+        if self.Gfs2Supported():
+            Print("Performing GFS2 control path stress tests.")
+            (retValControl, checkPointsControlGFS2,
+             totalCheckPointsControlGFS2) = \
+                self.gfs2_handler.ControlPathStressTests()
+
+            checkPointsControl += checkPointsControlGFS2
+            totalCheckPointsControl += totalCheckPointsControlGFS2
+
+            self.DisableClustering()
```

Comment:
@MarkSymsCtx,
I am not familiar with this feature, but have a question whether self.gfs2_handler.ControlPathStressTests() needs to put into "try" for exception capturing, and make sure self.DisableClustering() to be called in 'finally'?